public/app/core/components/NestedFolderPicker/NestedFolderPicker.test.tsx

```diff
@@ -174,14 +167,36 @@ describe('NestedFolderPicker', () => {
   });
 
   it('hides folders specififed by UID', async () => {
-    render(<NestedFolderPicker excludeUIDs={[folderA.item.uid]} onChange={mockOnChange} />);
+    render(<NestedFolderPicker excludeUIDs={[folderC.item.uid]} onChange={mockOnChange} />);
 
     // Open the picker and wait for children to load
     const button = await screen.findByRole('button', { name: 'Select folder' });
     await userEvent.click(button);
-    await screen.findByLabelText(folderB.item.title);
+    await screen.findByLabelText(folderA.item.title);
 
-    expect(screen.queryByLabelText(folderA.item.title)).not.toBeInTheDocument();
+    expect(screen.queryByLabelText(folderC.item.title)).not.toBeInTheDocument();
+  });
+
+  it('by default only shows items the user can edit', async () => {
+    render(<NestedFolderPicker onChange={mockOnChange} />);
+
+    const button = await screen.findByRole('button', { name: 'Select folder' });
+    await userEvent.click(button);
+    await screen.findByLabelText(folderA.item.title);
+
+    expect(screen.queryByLabelText(folderB.item.title)).not.toBeInTheDocument(); // folderB is not editable
+    expect(screen.queryByLabelText(folderC.item.title)).toBeInTheDocument(); // but folderC is
```

Comment:
nit, think we should just [use `getBy` when we're expecting it to be in the document](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence)

Reply:
So this would be 

```ts
expect(screen.queryByLabelText(folderB.item.title)).not.toBeInTheDocument(); // folderB is not editable
expect(screen.getByLabelText(folderC.item.title)).toBeInTheDocument(); // but folderC is
```

?

Reply:
yup 👍 then if the `getBy` ever fails, it will fail with a _much_ nicer error message